ipn/ipnlocal/serve.go

```diff
@@ -329,28 +331,29 @@ func (b *LocalBackend) HandleIngressTCPConn(ingressPeer tailcfg.NodeView, target
 	b.mu.Lock()
 	sc := b.serveConfig
 	b.mu.Unlock()
+	logf := logger.WithPrefix(b.logf, "handleIngress: ")
```

Comment:
this alloc on every call even if there's no log worth logging is a little sad.

maybe we need a map hanging off LocalBackend to cache these based on the prefix. `logf := b.prefixLogger("handleIngress")` or something.

Reply:
added a TODO